app/Filament/Actions/ImportAction.php

```diff
@@ -0,0 +1,267 @@
+<?php
+
+namespace App\Filament\Actions;
+
+use Filament\Forms;
+use App\Models\User;
+use App\Models\Import;
+use App\Jobs\ImportCsv;
+use App\Imports\Importer;
+use League\Csv\Statement;
+use Illuminate\Support\Arr;
+use Filament\Actions\Action;
+use App\Support\ChunkIterator;
+use Illuminate\Support\Facades\Bus;
+use League\Csv\Reader as CsvReader;
+use Illuminate\Support\Facades\Storage;
+use Filament\Notifications\Notification;
+use Illuminate\Filesystem\AwsS3V3Adapter;
+use App\Filament\Actions\ImportAction\ImportColumn;
+use Livewire\Features\SupportFileUploads\TemporaryUploadedFile;
+
+class ImportAction extends Action
+{
+    /**
+     * @var class-string<Importer>
+     */
+    protected string $importer;
+
+    protected ?string $job = null;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->modalHeading(fn (ImportAction $action): string => "Import {$action->getPluralModelLabel()}");
+
+        $this->modalSubmitActionLabel('Start import');
+
+        $this->groupedIcon('heroicon-m-arrow-up-tray');
+
+        $this->form(fn (ImportAction $action): array => array_merge([
+            Forms\Components\FileUpload::make('file')
```

Comment:
Small nit but, we prefer the direct `use` statements throughout the app to this partial. So that it is just `FileUpload` here.

Reply:
Done